db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1933,6 +1938,12 @@ Status DBImpl::RunManualCompaction(
   assert(!manual.in_progress);
   assert(HasPendingManualCompaction());
   RemoveManualCompaction(&manual);
+  // if the manual job is unscheduled, try schedule other jobs in case there's
+  // other pending jobs.
+  if (manual.status.IsIncomplete() &&
+      manual.status.subcode() == Status::SubCode::kManualCompactionPaused) {
+    MaybeScheduleFlushOrCompaction();
```

Comment:
It could be useful in the following situation:

step0. compaction pool is full
step1. exclusive manual compaction is scheduled
step2. 4 new flushes happens -> `unscheduled_compactions_` is bumped, but unable to schedule:
https://github.com/facebook/rocksdb/blob/d74468e348068b1d47e836c2c71141e670ae287b/db/db_impl/db_impl_compaction_flush.cc#L2441-L2443
step3. Unschedule the manual compaction here, but the `unscheduled_compactions_` won't be scheduled until the next event to trigger `MaybeScheduleFlushOrCompaction()`. So I think we should explicitly try schedule it again.

Reply:
Sounds good!

Reply:
I updated comment to mention that's for scheduling potential auto compaction which was blocked by exclusive manual compaction.